google/cloud/logging_v2/handlers/_helpers.py

```diff
@@ -108,54 +113,94 @@ def get_request_data_from_django():
     }
 
     # find trace id and span id
-    header = request.META.get(_DJANGO_TRACE_HEADER)
-    trace_id, span_id = _parse_trace_span(header)
+    header = request.META.get(_DJANGO_TRACEPARENT)
+    if header:
+        trace_id, span_id, trace_sampled = _parse_trace_parent(header)
+    else:
+        # w3 traceparent header not found. Check XCLOUD_TRACE_CONTEXT
+        header = request.META.get(_DJANGO_XCLOUD_TRACE_HEADER)
+        trace_id, span_id, trace_sampled = _parse_xcloud_trace(header)
+
+    return http_request, trace_id, span_id, trace_sampled
 
-    return http_request, trace_id, span_id
+
+def _parse_trace_parent(header):
+    """Given a w3 traceparent header, extract the trace and span ids.
+    For more information see https://www.w3.org/TR/trace-context/
+
+    Args:
+        header (str): the string extracted from the traceparent header
+            example: 00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01
+    Returns:
+        Tuple[Optional[dict], Optional[str], bool]:
+            The trace_id, span_id and trace_sampled extracted from the header
+            Each field will be None if header can't be parsed in expected format.
+    """
+    trace_id = span_id = None
+    trace_sampled = False
+    # see https://www.w3.org/TR/trace-context/ for W3C traceparent format
+    if header:
+        try:
+            VERSION_PART = r"(?!ff)[a-f\d]{2}"
```

Comment:
nit: right now the version **must be** `00`. other versions are considered invalid.

Reply:
How are you handling other versions? Just ignoring anything other than `00`?

It looks like [nodejs is assuming forward-compatibility](https://github.com/googleapis/nodejs-logging/blob/005cced4f8c1fe5f90496537e4731c2272946cb7/src/context.ts#L228), based on [this part of the spec](https://www.w3.org/TR/trace-context/#versioning-of-traceparent) ("The current version of this specification assumes that future versions of the traceparent header will be additive to the current one."). I'm fine with locking to version 00 for now though if we decide that's safer. Let me know if you have thoughts.